scripts/bootstrap.sh.in

```diff
@@ -167,6 +167,24 @@ check_ca_minion() {
     fi
 }
 
+check_minion_id() {
+    # Minion ID is used as the Kubernetes Node name, so it must follow the
+    # RFC1123 (https://tools.ietf.org/html/rfc1123).
+    # This means the name must:
+    #   - contain no more than 253 characters
+    #   - contain only lowercase alphanumeric characters, '-' or '.'
+    #   - start with an alphanumeric character
+    #   - end with an alphanumeric character
+    minion_id=$(get_salt_minion_id)
+
+    if ! [[ $minion_id =~ ^(([0-9a-z][0-9a-z.-]{0,251}[0-9a-z])|[0-9a-z])$ ]]; then
+        echo "Invalid Salt minion ID '$minion_id': The ID must be compliant" \
+             "with RFC1123, which means it must contain no more than 253" \
+             "characters, contain only lowercase alphanumeric characters, '-'" \
+             "or '.' and start and end with an alphanumeric character."
+        return 1
+    fi
+}
```

Comment:
To me, it should be done in `metalk8s_checks` salt execution module. We want to do this check for all nodes and not only for bootstrap

Reply:
Doing it here allow to fail early in the bootstrap process, maybe we could add it also in `metalk8s_checks` but since we deploy new node by adding Node manifest before even configuring Salt, it should fail at this moment.